models: rule updates for author, ASIN, abstract

ludmilamarian opened the pull request:
* Updates the rules for authors, external identifiers,
  abstract. (closes #179)

* Fixes author ID mapping and adds a new schema for ID: CDS;
  Adds default role to all authors.

* Fixes failing test cases for videos: affiliation is no
  longer present if the user leaves CERN.

Signed-off-by: Ludmila Marian <ludmila.marian@gmail.com>

kprzerwa commented on cds_dojson/marc21/fields/utils.py:
```diff
@@ -309,39 +309,79 @@ def _get_correct_video_contributor_role(role):
 
-def build_contributor(value):
+def build_contributor_books(value):
+    """Create the contributors for books"""
```
missing dot in the end of docstring

ludmilamarian replied:
I need to fix my editor, hopefully Travis will be happy as well :-)

kprzerwa commented on cds_dojson/marc21/fields/utils.py:
```diff
@@ -309,39 +309,79 @@ def _get_correct_video_contributor_role(role):
+    contributor = {
+        'ids': _extract_json_ids(value, 'schema') or None,
+        'full_name': value.get('name') or value.get('a'),
+        'email': value.get('email'),
```
do you clean somehow the values? for example strip of white spaces or perform a check if the value is present? Not checking it caused problems in keywords in videos, blank fields were migrated

ludmilamarian replied:
I added some cleaning, let me know if you think I should be doing even more

ludmilamarian commented:
This pull request **introduces 1 alert** when merging c5e46a13a76f1629b85d40d48e2ae7189d35a61f into 84e0d6915843fdb0fcb99f93e686985e5f90cf9a - [view on LGTM.com](https://lgtm.com/projects/g/CERNDocumentServer/cds-dojson/rev/pr-a5530876d3d18726666f7351dd482d9a7aa16e2c)

**new alerts:**

* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

kprzerwa commented on cds_dojson/schemas/records/books/book/base-v0.0.1.json:
```diff
@@ -162,10 +162,10 @@
               "default": "author",
```
default should have uppercase as in the enum below

ludmilamarian replied:
well spotted! should be fixed.

kprzerwa commented on cds_dojson/schemas/records/books/book/document-v0.0.1.json:
```diff
@@ -680,10 +704,10 @@
               "default": "author",
```
default should have first uppercase, as changed in the enum below

ludmilamarian replied:
fixed.

ludmilamarian commented:
@kprzerwa I have fixed your comments and added the note for python 3.6, can you make another pass? (maybe only if travis is green, otherwise I will take it anyway to recheck)

ntarocco commented on cds_dojson/marc21/fields/utils.py:
```diff
@@ -309,39 +309,79 @@ def _get_correct_video_contributor_role(role):
-def _extract_json_ids(info):
+def _get_correct_books_contributor_role(role):
+    """Clean up roles."""
+    tranlations = {
```
very very minor, typo: tranlations -> translations

ludmilamarian replied:
well spotted, thanks! Fixed in all 4 places.
